app/assets/javascripts/discourse/app/controllers/avatar-selector.js

```diff
@@ -21,11 +21,40 @@ export default Controller.extend(ModalFunctionality, {
     "siteSettings.selectable_avatars"
   )
   selectableAvatars(enabled, list) {
-    if (enabled) {
+    if (enabled !== "none") {
       return list ? list.split("|") : [];
     }
   },
 
+  @discourseComputed("siteSettings.selectable_avatars_enabled")
+  showSelectableAvatars(enabled) {
+    return enabled !== "none";
+  },
+
+  @discourseComputed("siteSettings.selectable_avatars_enabled")
+  showAvatarUploader(selectableAvatars) {
+    if (!selectableAvatars || selectableAvatars === "none") {
+      return true;
+    }
+    if (selectableAvatars === "restrict_all") {
+      return false;
+    }
+    if (selectableAvatars?.startsWith("restrict_tl")) {
+      const restrictedTl = parseInt(
+        selectableAvatars.replace("restrict_tl", ""),
+        10
+      );
+      return (
+        this.user.admin ||
+        this.user.moderator ||
+        this.user.trust_level > restrictedTl
+      );
+    }
+    if (selectableAvatars === "restrict_nonstaff") {
+      return this.user.admin || this.user.moderator;
+    }
```

Comment:
I think this is one of the times a `switch` is clearer as it lists all the possible values of the enum.

```suggestion
    switch (selectableAvatars) {
      case "restrict_all":
        return false;
      case "restrict_tl1":
      case "restrict_tl2":
      case "restrict_tl3":
        const restrictedTl = parseInt(
          selectableAvatars.replace("restrict_tl", ""),
          10
        );
        return (
          this.user.admin ||
          this.user.moderator ||
          this.user.trust_level > restrictedTl
        );
      case "restrict_nonstaff":
        return this.user.admin || this.user.moderator;
      case "none":
      default:
        return true;
    }
```